Feat: Implement actor gallery screen

khaledeid1k opened the pull request.

nourelhodaahmed commented on presentation/src/main/java/com/moscow/cineverse/screen/cast_details_gallery/ActorGalleryScreen.kt:
```diff
@@ -0,0 +1,127 @@
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun ActorGalleryScreen(
```
don't you need the actor id as a param here ?

khaledeid1k committed:
feat: Implement actor gallery screen

This commit introduces the `ActorGalleryScreen` which displays a gallery of photos for a specific actor.

**Key components:**
- `ActorGalleryViewModel`: Manages the state and logic for fetching actor photos.
- `ActorGalleryState`: Represents the UI state (loading, error, photos list).
- `ActorGalleryEvents`: Defines possible events (currently empty).
- `ActorGalleryInteractionListener`: Handles user interactions like refresh and back button clicks.
- `ActorGalleryScreen`: Composable function that observes the ViewModel and renders the UI.
- `ActorGalleryContent`: Composable function responsible for the layout and displaying different states (loading, error, content).

**Functionality:**
- Fetches actor photos using `GetActorGallery` use case.
- Displays a loading indicator while fetching data.
- Shows an error message with a retry button on failure.
- Renders the actor photos in a `CastGallery` component.
- Includes a `MovieAppBar` with a title and back button.

The `ViewModelModule` has been updated to include `ActorGalleryViewModel`.

nourelhodaahmed committed:
feat: get actor id and actor name from navigation

nourelhodaahmed committed:
Merge branch 'feature/105-implement-cast-details-screen' into feature/112-implement-cast-details-screen_actor_gallery

# Conflicts:
#	app/src/main/java/com/moscow/cineverse/MainActivity.kt

nourelhodaahmed committed:
refactor: remove unused import

nourelhodaahmed committed:
refactor: add params to view model instead of loading data in launch effect
